sources/packages/backend/libs/utilities/src/system-users/system-users.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { User } from "@sims/sims-db";
+import { Repository } from "typeorm";
+import { SERVICE_ACCOUNT_DEFAULT_USER_EMAIL } from "../system-configurations-constants";
+import { SystemUser, SystemUserDetails } from "./system-users.models";
+
+@Injectable()
+export class SystemUsersService {
+  constructor(
+    @InjectRepository(User)
+    private userRepo: Repository<User>,
+  ) {}
+
+  /**
+   * Get system user. If not found create a new
+   * system user and return the user.
+   * @return system user.
+   */
+  private async getSystemUser(): Promise<User> {
+    const existingUser = await this.userRepo.findOne({
+      select: {
+        id: true,
+      },
+      where: {
+        userName: SystemUser.UserName,
+      },
+    });
+
+    if (existingUser) {
+      return existingUser;
+    }
+
+    // Create new system user if not exists.
+    const user = new User();
+    user.userName = SystemUser.UserName;
+    user.firstName = null;
+    user.lastName = SystemUser.LastName;
+    user.email = SERVICE_ACCOUNT_DEFAULT_USER_EMAIL;
+    return this.userRepo.save(user);
+  }
+
+  /**
+   * Get system user details from the cache.
+   * If not found will create a new system user
+   * and return the user details.
+   * @return system user details.
+   */
+  async systemUser(): Promise<SystemUserDetails> {
```

Comment:
I do not see an issue in return the User entity model here. Thoughts?

Reply:
mm, I was thinking that as per future demand we can add the property in the SystemUserDetails

Reply:
I belive that either way is ok. The suggestion was to try to avoid any transformation while consuming it.
For instance, when assigning the audit user as below.
```ts
const myDatabaseModel = new MyDatabaseModel();
myDatabaseModel.creator = await this.systemUsersService.systemUser();
```
instead of
```ts
const myDatabaseModel = new MyDatabaseModel();
myDatabaseModel.creator = { id: await this.systemUsersService.systemUser().id } as User;
```